src/libsync/account.cpp

```diff
@@ -163,15 +163,7 @@ QString Account::displayName() const
 
 QString Account::userIdAtHostWithPort() const
 {
-    const auto credentialsUserSplit = credentials() ? credentials()->user().split(QLatin1Char('@')) : QStringList{};
-
-    if (credentialsUserSplit.isEmpty()) {
-        return {};
-    }
-
-    const auto userName = credentialsUserSplit.first();
-
-    QString dn = QStringLiteral("%1@%2").arg(userName, _url.host());
+    QString dn = QStringLiteral("%1@%2").arg(_davUser, _url.host());
```

Comment:
in case _davUser will contain a string with '@' (like user@example.cloud.com), you will have an incorrect value in `dn` -> `user@example.cloud.com@example.cloud.com `

Reply:
as far as I know, the email address would be in `_credentials->user()` but not `Account::_davUser`